test/TestProjects/ReferenceTypes/Generated/Models/CheckNameAvailabilityRequest.cs

```diff
@@ -22,17 +21,17 @@ public CheckNameAvailabilityRequest()
 
         /// <summary> Initializes a new instance of CheckNameAvailabilityRequest. </summary>
         /// <param name="name"> The name of the resource for which availability needs to be checked. </param>
-        /// <param name="type"> The resource type. </param>
+        /// <param name="checkNameAvailabilityRequestType"> The resource type. </param>
         [SerializationConstructor]
-        internal CheckNameAvailabilityRequest(string name, ResourceType type)
+        internal CheckNameAvailabilityRequest(string name, string checkNameAvailabilityRequestType)
         {
             Name = name;
-            Type = type;
+            CheckNameAvailabilityRequestType = checkNameAvailabilityRequestType;
         }
 
         /// <summary> The name of the resource for which availability needs to be checked. </summary>
         public string Name { get; set; }
         /// <summary> The resource type. </summary>
-        public ResourceType Type { get; set; }
+        public string CheckNameAvailabilityRequestType { get; set; }
```

Comment:
This model only has name and type, no id. Therefore, the result of the auto-renaming is not what we want. To correct it, we can set x-ms-client-name to `ResourceType`. The result will become `public ResourceType ResourceType`.